File: BotsSDK/korebotsdklib/src/main/java/kore/botssdk/models/BrandingFooterModel.java
package kore.botssdk.models;

import java.io.Serializable;

import kore.botssdk.utils.StringUtils;

public class BrandingFooterModel implements Serializable {
    private String bg_color;
    private String layout;
    private String style;
    private String icons_color;
    private BrandingFooterComposeBarModel compose_bar;
    private BrandingFooterButtonsModel buttons;

    public BrandingFooterModel updateWith(BrandingFooterModel configModel) {
        bg_color = !StringUtils.isNullOrEmpty(configModel.bg_color) ? configModel.bg_color : bg_color;
        layout = !StringUtils.isNullOrEmpty(configModel.layout) ? configModel.layout : layout;
        style = !StringUtils.isNullOrEmpty(configModel.style) ? configModel.style : style;
        icons_color = !StringUtils.isNullOrEmpty(configModel.icons_color) ? configModel.icons_color : icons_color;
        compose_bar = compose_bar != null && configModel.compose_bar != null ? compose_bar.updateWith(configModel.compose_bar) : compose_bar;
        buttons = buttons != null && configModel.buttons != null ? buttons.updateWith(configModel.buttons) : buttons;
        return this;
    }

    public String getBg_color() {
        return bg_color;
    }

    public String getStyle() {
        return style;
    }

    public BrandingFooterButtonsModel getButtons() {
        return buttons;
    }

    public BrandingFooterComposeBarModel getCompose_bar() {
        return compose_bar;
    }

    public String getLayout() {
        return layout;
    }

    public void setIcons_color(String icons_color) {
        this.icons_color = icons_color;
    }

    public String getIcons_color() {
        return icons_color;
    }

    public void setBg_color(String bg_color) {
        this.bg_color = bg_color;
    }
}

File: BotSdk/botclient/src/main/java/com/kore/network/api/responsemodels/branding/BrandingFooterModel.kt
package com.kore.network.api.responsemodels.branding

import com.google.gson.annotations.SerializedName

data class BrandingFooterModel(
    @SerializedName("bg_color")
    val bgColor: String? = null,
    @SerializedName("layout")
    val layout: String? = null,
    @SerializedName("style")
    val style: String? = null,
    @SerializedName("compose_bar")
    val composeBar: BrandingFooterComposeBarModel? = null,
    @SerializedName("icons_color")
    var iconsColor: String? = null,
    @SerializedName("buttons")
    val buttons: BrandingFooterButtonsModel? = null
)

fun BrandingFooterModel.updateWith(configModel: BrandingFooterModel): BrandingFooterModel {
    return this.copy(
        bgColor = configModel.bgColor?.ifEmpty { this.bgColor } ?: bgColor,
        style = configModel.style?.ifEmpty { this.style } ?: style,
        iconsColor = configModel.iconsColor?.ifEmpty { this.iconsColor } ?: iconsColor,
        layout = configModel.layout?.ifEmpty { this.layout } ?: layout,
        buttons = this.buttons?.updateWith(configModel.buttons) ?: this.buttons,
    )
}